### PR TITLE
Nixpkgs upgrade to NixOS 24.11

### PR DESCRIPTION
**Pull request by Mic92**
# Motivation

In preparation to NixOS 24.11 this updates nixpkgs to unstable until we can switch to the branch off branch. As a side effect this should also help with https://github.com/NixOS/nix/issues/11546
So far only the x86_64-linux build has been tested. 

# Context

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   description = "The purely functional package manager";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
+  #inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  # TODO: Switch to nixpkgs-unstable once this is in: https://nixpkgs-tracker.ocfox.me/?pr=344090
+  inputs.nixpkgs.url = "github:Mic92/nixpkgs/python3-netbsd";
```
Comment (Mic92) [inline flake.nix:6]: cc @Ericson2314 https://github.com/NixOS/nixpkgs/pull/344157

**Comment by Mic92** [comment]
Bash no longer evals on windows/mingw?

**Comment by Mic92** [comment]
nix-static on macOS is still broken:

```
checking derivation packages.aarch64-darwin.nix-static...
error:
       … while checking flake output 'packages'
         at /nix/store/lsh2kjgn0hhkas0wiazza4xdpllxfavr-source/flake.nix:221:7:
          220|
          221|       packages = forAllSystems (system:
             |       ^
          222|         { # Here we put attributes that map 1:1 into packages.<system>, ie

       … while checking the derivation 'packages.aarch64-darwin.nix-static'
         at /nix/store/lsh2kjgn0hhkas0wiazza4xdpllxfavr-source/flake.nix:270:15:
          269|               "${pkgName}" = nixpkgsFor.${system}.native.nixComponents.${pkgName};
          270|               "${pkgName}-static" = nixpkgsFor.${system}.static.nixComponents.${pkgName};
             |               ^
          271|             }

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'nix-static-aarch64-apple-darwin-2.25.0pre20240924_03afb39'
         whose name attribute is located at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute '__impureHostDeps' of derivation 'nix-static-aarch64-apple-darwin-2.25.0pre20240924_03afb39'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:451:7:
          450|       __propagatedSandboxProfile = unique (computedPropagatedSandboxProfile ++ [ propagatedSandboxProfile ]);
          451|       __impureHostDeps = computedImpureHostDeps ++ computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps ++ __impureHostDeps ++ stdenv.__extraImpureHostDeps ++ [
             |       ^
          452|         "/dev/zero"

       … while calling the 'foldl'' builtin
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:441:11:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |           ^
          442|             allPropagatedDependencies);

       … while calling the 'concatMap' builtin
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:441:19:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                   ^
          442|             allPropagatedDependencies);

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:441:30:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                              ^
          442|             allPropagatedDependencies);

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:441:37:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                                     ^
          442|             allPropagatedDependencies);

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:326:13:
          325|       (map (drv: getDev drv.__spliced.hostHost or drv) (checkDependencyList "depsHostHostPropagated" depsHostHostPropagated))
          326|       (map (drv: getDev drv.__spliced.hostTarget or drv) (checkDependencyList "propagatedBuildInputs" propagatedBuildInputs))
             |             ^
          327|     ]

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:326:18:
          325|       (map (drv: getDev drv.__spliced.hostHost or drv) (checkDependencyList "depsHostHostPropagated" depsHostHostPropagated))
          326|       (map (drv: getDev drv.__spliced.hostTarget or drv) (checkDependencyList "propagatedBuildInputs" propagatedBuildInputs))
             |                  ^
          327|     ]

       … while calling 'getOutput'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/attrsets.nix:1797:23:
         1796|   */
         1797|   getOutput = output: pkg:
             |                       ^
         1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified

       … while evaluating a branch condition
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/attrsets.nix:1798:5:
         1797|   getOutput = output: pkg:
         1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
             |     ^
         1799|       then pkg.${output} or pkg.out or pkg

       … in the left operand of the OR (||) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/attrsets.nix:1798:32:
         1797|   getOutput = output: pkg:
         1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
             |                                ^
         1799|       then pkg.${output} or pkg.out or pkg

       … in the argument of the not operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/attrsets.nix:1798:10:
         1797|   getOutput = output: pkg:
         1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
             |          ^
         1799|       then pkg.${output} or pkg.out or pkg

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:326:25:
          325|       (map (drv: getDev drv.__spliced.hostHost or drv) (checkDependencyList "depsHostHostPropagated" depsHostHostPropagated))
          326|       (map (drv: getDev drv.__spliced.hostTarget or drv) (checkDependencyList "propagatedBuildInputs" propagatedBuildInputs))
             |                         ^
          327|     ]

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/lists.nix:334:29:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                             ^
          335|

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/lists.nix:334:32:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                                ^
          335|

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:281:15:
          280|     imap1
          281|       (index: dep:
             |               ^
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep

       … while evaluating a branch condition
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:282:9:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |         ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … in the left operand of the OR (||) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:282:69:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |                                                                     ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … in the left operand of the OR (||) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:282:44:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |                                            ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … in the left operand of the OR (||) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:282:29:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |                             ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:282:12:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |            ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … while calling 'isDerivation'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/attrsets.nix:1283:5:
         1282|   isDerivation =
         1283|     value: value.type or null == "derivation";
             |     ^
         1284|

       … while calling the 'elemAt' builtin
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/lists.nix:334:43:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                                           ^
          335|

       … while evaluating the attribute 'boost.override'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/top-level/all-packages.nix:19206:3:
        19205|
        19206|   boost = boost181;
             |   ^
        19207|

       … while evaluating the attribute 'boost181'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/default.nix:24:3:
           23|   boost180 = makeBoost ./1.80.nix;
           24|   boost181 = makeBoost ./1.81.nix;
             |   ^
           25|   boost182 = makeBoost ./1.82.nix;

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/default.nix:24:14:
           23|   boost180 = makeBoost ./1.80.nix;
           24|   boost181 = makeBoost ./1.81.nix;
             |              ^
           25|   boost182 = makeBoost ./1.82.nix;

       … while calling 'makeBoost'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/default.nix:8:15:
            7| let
            8|   makeBoost = file:
             |               ^
            9|     lib.fix (self:

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/default.nix:9:5:
            8|   makeBoost = file:
            9|     lib.fix (self:
             |     ^
           10|       callPackage file {

       … while calling 'fix'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/fixed-points.nix:93:5:
           92|   fix =
           93|     f:
             |     ^
           94|     let

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/fixed-points.nix:95:11:
           94|     let
           95|       x = f x;
             |           ^
           96|     in

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/default.nix:9:14:
            8|   makeBoost = file:
            9|     lib.fix (self:
             |              ^
           10|       callPackage file {

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/default.nix:10:7:
            9|     lib.fix (self:
           10|       callPackage file {
             |       ^
           11|         boost-build = boost-build.override {

       … while calling 'callPackageWith'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:212:35:
          211|   */
          212|   callPackageWith = autoArgs: fn: args:
             |                                   ^
          213|     let

       … while calling a functor (an attribute set with a '__functor' attribute)
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:264:13:
          263|     in if missingArgs == {}
          264|        then makeOverridable f allArgs
             |             ^
          265|        # This needs to be an abort so it can't be caught with `builtins.tryEval`,

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/trivial.nix:957:7:
          956|     { # TODO: Should we add call-time "type" checking like built in?
          957|       __functor = self: f;
             |       ^
          958|       __functionArgs = args;

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:136:17:
          135|     in
          136|     mirrorArgs (origArgs:
             |                 ^
          137|     let

       … while evaluating a branch condition
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:148:7:
          147|     in
          148|       if isAttrs result then
             |       ^
          149|         result // {

       … while calling the 'isAttrs' builtin
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:148:10:
          147|     in
          148|       if isAttrs result then
             |          ^
          149|         result // {

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:138:16:
          137|     let
          138|       result = f origArgs;
             |                ^
          139|

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/1.81.nix:1:1:
            1| { callPackage, fetchurl, ... } @ args:
             | ^
            2|

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/1.81.nix:3:1:
            2|
            3| callPackage ./generic.nix (args // rec {
             | ^
            4|   version = "1.81.0";

       (7 duplicate frames omitted)

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/generic.nix:1:1:
            1| { lib, stdenv, icu, expat, zlib, bzip2, zstd, xz, python ? null, fixDarwinDylibNames, libiconv, libxcrypt
             | ^
            2| , makePkgconfigItem

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/development/libraries/boost/generic.nix:102:1:
          101|
          102| stdenv.mkDerivation {
             | ^
          103|   pname = "boost";

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/adapters.nix:23:83:
           22|   # Wrap the original `mkDerivation` providing extra args to it.
           23|   extendMkDerivationArgs = old: f: withOldMkDerivation old (_: mkDerivationSuper: args:
             |                                                                                   ^
           24|     (mkDerivationSuper args).overrideAttrs f);

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/adapters.nix:24:6:
           23|   extendMkDerivationArgs = old: f: withOldMkDerivation old (_: mkDerivationSuper: args:
           24|     (mkDerivationSuper args).overrideAttrs f);
             |      ^
           25|

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/adapters.nix:23:83:
           22|   # Wrap the original `mkDerivation` providing extra args to it.
           23|   extendMkDerivationArgs = old: f: withOldMkDerivation old (_: mkDerivationSuper: args:
             |                                                                                   ^
           24|     (mkDerivationSuper args).overrideAttrs f);

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/adapters.nix:24:6:
           23|   extendMkDerivationArgs = old: f: withOldMkDerivation old (_: mkDerivationSuper: args:
           24|     (mkDerivationSuper args).overrideAttrs f);
             |      ^
           25|

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/adapters.nix:142:82:
          141|     extraBuildInputs = [ pkgs.buildPackages.darwin.CF ];
          142|     mkDerivationFromStdenv = withOldMkDerivation old (stdenv: mkDerivationSuper: args:
             |                                                                                  ^
          143|     (mkDerivationSuper args).overrideAttrs (prevAttrs: {

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/adapters.nix:143:5:
          142|     mkDerivationFromStdenv = withOldMkDerivation old (stdenv: mkDerivationSuper: args:
          143|     (mkDerivationSuper args).overrideAttrs (prevAttrs: {
             |     ^
          144|       NIX_CFLAGS_LINK = toString (prevAttrs.NIX_CFLAGS_LINK or "")

       … while calling anonymous lambda
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:100:8:
           99|     mkDerivationSimple
          100|       (f0:
             |        ^
          101|         let

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:110:11:
          109|         in
          110|           makeDerivationExtensible (self: attrs // (if builtins.isFunction f0 || f0?__functor then f self attrs else f0)))
             |           ^
          111|       attrs;

       … while calling 'makeDerivationExtensible'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:77:30:
           76|   # Based off lib.makeExtensible, with modifications:
           77|   makeDerivationExtensible = rattrs:
             |                              ^
           78|     let

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:92:9:
           91|       finalPackage =
           92|         mkDerivationSimple overrideAttrs args;
             |         ^
           93|

       … while calling 'mkDerivationSimple'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:516:1:
          515| # * [package attributes]: https://nixos.org/manual/nix/stable/glossary#package-attribute-set
          516| {
             | ^
          517|

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:596:1:
          595|
          596| extendDerivation
             | ^
          597|   validity.handled

       … while calling 'extendDerivation'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:339:43:
          338|   */
          339|   extendDerivation = condition: passthru: drv:
             |                                           ^
          340|     let

       … in the left operand of the update (//) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:364:20:
          363|       outputsList = map outputToAttrListElement outputs;
          364|     in commonAttrs // {
             |                    ^
          365|       drvPath = assert condition; drv.drvPath;

       … in the left operand of the update (//) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/customisation.nix:343:25:
          342|
          343|       commonAttrs = drv // (listToAttrs outputsList) //
             |                         ^
          344|         ({ all = map (x: x.value) outputsList; }) // passthru;

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:645:4:
          644|    passthru)
          645|   (derivation (derivationArg // optionalAttrs envIsExportable checkedEnv));
             |    ^
          646|

       … in the right operand of the update (//) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:645:30:
          644|    passthru)
          645|   (derivation (derivationArg // optionalAttrs envIsExportable checkedEnv));
             |                              ^
          646|

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:645:33:
          644|    passthru)
          645|   (derivation (derivationArg // optionalAttrs envIsExportable checkedEnv));
             |                                 ^
          646|

       … while calling 'optionalAttrs'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/attrsets.nix:1352:5:
         1351|     cond:
         1352|     as:
             |     ^
         1353|     if cond then as else {};

       … in the condition of the assert statement
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:581:5:
          580|       "When using structured attributes, `env` must be an attribute set of environment variables.";
          581|     assert assertMsg (overlappingNames == [ ])
             |     ^
          582|       "The `env` attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping:\n${errors}";

       … from call site
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/pkgs/stdenv/generic/make-derivation.nix:581:12:
          580|       "When using structured attributes, `env` must be an attribute set of environment variables.";
          581|     assert assertMsg (overlappingNames == [ ])
             |            ^
          582|       "The `env` attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping:\n${errors}";

       … while calling 'assertMsg'
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/asserts.nix:41:5:
           40|     pred:
           41|     msg:
             |     ^
           42|     pred || builtins.throw msg;

       … in the right operand of the OR (||) operator
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/asserts.nix:42:10:
           41|     msg:
           42|     pred || builtins.throw msg;
             |          ^
           43|

       … while calling the 'throw' builtin
         at /nix/store/qar7dqcwfh7zgp49gik8wm3hzysw5llw-source/lib/asserts.nix:42:13:
           41|     msg:
           42|     pred || builtins.throw msg;
             |             ^
           43|

       error: The `env` attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping:
         - NIX_CFLAGS_LINK: in `env`: "-headerpad_max_install_names"; in derivation arguments: ""
```

**Comment by reckenrode** [comment]
> nix-static on macOS is still broken:

This should be fixed by the Darwin refactor work that’s coming (see: https://discourse.nixos.org/t/on-the-future-of-darwin-sdks-or-how-you-can-stop-worrying-and-put-the-sdk-in-build-inputs/50574). I’m working through conflicts after rebasing on current staging. I have a few issues left, but I’m hopeful I can get the draft PR opened tonight to start the review while I work through them. The goal is to get it into the next staging-next cycle and be part of the 24.11 release.

**Comment by edolstra** [comment]
Hydra jobset: https://hydra.nixos.org/jobset/nix/pr-11566

**Comment by Mic92** [comment]
TODO: fix whatever is broken after hydra eval

**Comment by reckenrode** [comment]
fyi, https://github.com/NixOS/nixpkgs/pull/346043 is in master now. Darwin static compilation (both aarch64 and x86_64) should work now.

**Comment by Mic92** [comment]
Ok. Github looks good. Let's see what hydra says: https://hydra.nixos.org/eval/1810385

**Comment by Mic92** [comment]
@mergify rebase

**Review by roberth — APPROVED** [review]

**Comment by roberth** [comment]
Saw one or two failed tests on Hydra, but 24.05 is not relevant anymore, so let's fix them on master instead.

File: flake.nix [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
   description = "The purely functional package manager";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/release-24.11";
```
Comment (cole-h) [inline flake.nix:4]: Out of curiosity, why is this using the master-branch-equivalent of 24.11 instead of, say, `nixpkgs-24.11` or `nixos-24.11`?

**Pull request merged**